### PR TITLE
coverage: add UI test coverage reporting

### PR DESCRIPTION
**Pull request by mikemorris**
This is an experiment in adding separate UI test coverage reporting using CodeCov [flags](https://docs.codecov.io/docs/flags) functionality - it looks like the underlying infrastructure is [already set up](https://github.com/hashicorp/consul/blob/master/ui-v2/package.json#L26), just need to get it reporting in CI...

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -509,6 +510,10 @@ jobs:
       - run:
           working_directory: ui-v2
           command: node_modules/ember-cli/bin/ember exam --split=$CIRCLE_NODE_TOTAL --partition=`expr $CIRCLE_NODE_INDEX + 1` --path dist --silent -r xunit
```
Comment (mikemorris) [inline .circleci/config.yml]: I'm not familiar with `ember exam`, is there additional config that would be needed to output a code coverage report from here similar to the [`test:coverage`](https://github.com/hashicorp/consul/blob/master/ui-v2/package.json#L26) npm script @johncowen? (the CodeCov upload doesn't seem to be uploading any frontend test coverage data currently)

File: codecov.yml [hunk 2]
```diff
@@ -28,5 +28,8 @@ coverage:
 comment: false
 
 # https://docs.codecov.io/docs/flags
```
Comment (mikemorris) [inline codecov.yml]: Unfortunately, this functionality appears to be limited in the full UI currently due to performance issues https://docs.codecov.io/docs/flags#section-flags-in-the-codecov-ui

**Comment by johncowen** [comment]
hey @mikemorris , I know we've chatted about this offline, but thought I'd note here also. I'm pretty sure there is a little bit of work to do integrating `ember-cli-code-coverage` here. I can't remember exactly but the this plugin uses a HTTP request to report the coverage - which unfortunately gets caught by the mock XMLHttpRequest we use to be able to mock the Consul API.

When I've looked at our coverage in the past, I've manually disabled the mock XMLHttpRequest and then ran the coverage report on our unit tests only to get an idea of our coverage.

It would be great to get this into CI, so we'd need to look at allowing that single request that the report needs to use the native XMLHttpRequest instead of the mock one.

A little tangent here, following out refactor of the data layer in the UI, we now have control of how we make HTTP requests in the UI - and I have a vague plan to move away from using XMLHttpRequest on the main thread. This would mean we wouldn't have to mock the native XMLHttpRequest and therefore the reporting functionality of `ember-cli-code-coverage` would just work as normal. This is a bit more of a change though, and I wouldn't imagine it happening soon.

**Comment by johncowen** [comment]
Hey @mikemorris ,

We can probably close this in favour of the other PR referenced here right?

**Comment by i0rek** [comment]
I am closing this now, since I want this PR counter to go down. If it shouldn't have been closed, feel free to reopen!

**Pull request closed without merging**